### DISCUSSION TITLE
Trigger recompilation of single page

### DISCUSSION THREAD

**Discussion by ovp87** [tighten/jigsaw#599] [Q&A] [answered]
Is it currently possible to trigger the recompilation of a single page?

Given a situation where the application is quite large, and the static site generation takes some time, it would be nice to be able to trigger a recompile of just that one page.

**Comment by bakerkretzmar** [comment, marked as answer]
No, there isn't currently a way to do that. It's pretty complicated because pages can depend on each other, for example if you have pagination or "previous" and "next" collection items where updating one item could require re-rendering several pages.

How long does your Jigsaw build take? I'm happy to talk about it more and see if there are other ways to speed it up.

**Reply by ovp87** [reply]
Gotcha. 

Very understandable, I'm not currently using it at all (beyond simple testing), not sure my use case is the best fit for it, but what i was envisioning was something along these lines:

An Ecommerce saas; 
- Where the number of products could easily number in the thousands.
- If the merchant wants to, also supports multiple languages, further increasing the potential max-page count.

What i pictured was essentially recompiling only the required pages when a product was updated, either through some queue system, or just by listening to some eloquent event, to keep recompile times to a minimum.

**Reply by bakerkretzmar** [reply]
That makes sense, honestly I doubt that kind of partial rebuild would be possible with Jigsaw, but depending how your content is structured it might still be pretty quick to just rebuild the whole site.

That being said, your use case does sound like something that might be better served by a full-on app with a database 😅

**Reply by ovp87** [reply]
Yeah, i agree, not what it was made for. Thanks for an awesome package all the same 👍
